frontend/src/pages/TrainingReports/components/SessionCard.js

```diff
@@ -0,0 +1,132 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import { DECIMAL_BASE } from '@ttahub/common';
+import { Link } from 'react-router-dom';
+import UserContext from '../../../UserContext';
+import {
+  InProgress,
+  Closed,
+  NoStatus,
+  Pencil,
+} from '../../../components/icons';
+import { canEditOrCreateSessionReports } from '../../../permissions';
+import './SessionCard.scss';
+
+function SessionCard({
+  eventId,
+  session,
+  expanded,
+}) {
+  const { user } = useContext(UserContext);
+
+  const {
+    sessionName,
+    startDate,
+    endDate,
+    objective,
+    objectiveSupportType,
+    objectiveTopics,
+    objectiveTrainers,
+    status,
+    regionId,
+  } = session.data;
+
+  const hasEditPermissions = canEditOrCreateSessionReports(user, parseInt(regionId, DECIMAL_BASE));
+
+  const getSessionDisplayStatusText = () => {
+    if (status && (status === 'In progress' || status === 'Complete')) {
+      return status;
+    }
+    return 'Needs status';
+  };
+

```

Comment:
this could probably be a switch statement with a default cause of 'Needs status' for readability, e.g.

```typescript
switch (status) {
    case 'In progress':
    case 'Complete':
      return status;
    default:
      return 'Needs status';
  }
```

Reply:
I have updated this.